product-service/src/catalogBatchProcess.js

```diff
@@ -0,0 +1,28 @@
+const AWS = require('aws-sdk');
+const createProduct = require('../db/createProduct');
+const { REGION } = require('./values');
+
+module.exports = async event => {
+  const sns = new AWS.SNS({ region: REGION });
+
+  for (const record of event.Records) {
+    try {
+      const product = JSON.parse(record.body);
+
+      const { title, artist, description, price: stringPrice, coverurl, count: stringCount } = product;
+      const price = parseInt(stringPrice);
+      const count = parseInt(stringCount);
+
+      await createProduct(title, artist, description, price, coverurl, count)
+
+      console.log(`product was added: ${JSON.stringify(product)}`);
+      await sns.publish({
```

Comment:
products are sent one by one, but should be sent in batches once all products from event are processed

Reply:
I mean task does not state it clearly, but that would be a logical way of processing a batch of 5 received products

Reply:
yep, fixed (batch size updated to 3) and notification sending after whole batch processing with the corresponding email
<img width="551" alt="Screenshot 2020-11-26 at 19 28 59" src="https://user-images.githubusercontent.com/11299517/100374690-b42d8900-301d-11eb-9878-e2b4ef145a10.png">